components/templates/src/global_fns/mod.rs

```diff
@@ -279,15 +281,31 @@ impl TeraFn for GetImageMeta {
         if !src_path.exists() {
             return Err(format!("`get_image_metadata`: Cannot find path: {}", path).into());
         }
-        let img = image::open(&src_path)
-            .map_err(|e| Error::chain(format!("Failed to process image: {}", path), e))?;
+        let (height, width) = image_dimensions(&src_path)?;
         let mut map = tera::Map::new();
-        map.insert(String::from("height"), Value::Number(tera::Number::from(img.height())));
-        map.insert(String::from("width"), Value::Number(tera::Number::from(img.width())));
+        map.insert(String::from("height"), Value::Number(tera::Number::from(height)));
+        map.insert(String::from("width"), Value::Number(tera::Number::from(width)));
         Ok(Value::Object(map))
     }
 }
 
+// Try to read the image dimensions for a given image
+fn image_dimensions(path: &PathBuf) -> Result<(u32, u32)> {
```

Comment:
that's a place where anonymous structs would be very helpful, no need to member which u32 is height and width.

Reply:
True. We could add a `Dimensions` struct instead, but I think that would unnecessarily increase the complexity.